src/playfair.rs

```diff
@@ -0,0 +1,429 @@
+//! encountered 'J' characters with 'I'.
+
+use common::{alphabet, alphabet::Alphabet, cipher::Cipher};
+
+const PLAYFAIR_ALPHABET: &'static str = "ABCDEFGHIKLMNOPQRSTUVWXYZ";
+const PLAYFAIR_FIX_CHAR: char = 'X';
+
+// Playfair Cipher Modes
+enum CipherMode {
+    // Encrypt Mode
+    ENCRYPT,
+    // Decrypt Mode
+    DECRYPT,
+}
+
+// PLayfair Bigram
+type Bigram = (char, char);
+
+/// Apply rule 1 (bigrams).
+///
+/// "If both letters are the same (or only one letter is left), add an 'X'
+/// after the first letter. Encrypt the new pair and continue. Some variants
+/// of Playfair use 'Q' instead of 'X', but any letter, itself uncommon as a
+/// repeated pair, will do."
+///
+/// [Reference](https://en.wikipedia.org/wiki/Playfair_cipher#Description)
+///
+/// # Panics
+///
+/// Panics if message contain non-alpha characters.
+fn bigram<S: AsRef<str>>(message: S) -> Result<Vec<Bigram>, &'static str> {
+    if message.as_ref().contains(char::is_whitespace) {
+        panic!("Message contains whitespace");
+    }
+    if !alphabet::STANDARD.is_valid(message.as_ref()) {
+        panic!("Message must only consist of alphabetic characters");
+    }
+
+    let mut iter = message.as_ref().chars().peekable();
+    let mut bigrams: Vec<Bigram> = Vec::new();
+    loop {
+        let first: char;
+        if let Some(x) = iter.next() {
+            first = x;
+        } else {
+            // No characters remaining -- done
+            break;
+        }
+
+        // Handle repeats
+        if let Some(y) = iter.peek() {
+            if *y == first {
+                bigrams.push((first, PLAYFAIR_FIX_CHAR));
+                continue;
+            }
+        }
+
+        if let Some(y) = iter.next() {
+            bigrams.push((first, y));
+        } else {
+            // Handle odd number of characters
+            bigrams.push((first, PLAYFAIR_FIX_CHAR));
+        }
+    }
+    Ok(bigrams)
+}
+
+/// Apply rule 2 (Row) or rule 3 (Column).
+///
+/// "If the letters appear on the same row of your table, replace them
+/// with the letters to their immediate right respectively (wrapping
+/// around to the left side of the row if a letter in the original pair
+/// was on the right side of the row)."
+///
+/// "If the letters appear on the same column of your table, replace them
+/// with the letters immediately below respectively (wrapping around to the
+/// top side of the column if a letter in the original pair was on the
+/// bottom side of the column)."
+///
+/// [Reference](https://en.wikipedia.org/wiki/Playfair_cipher#Description)
+fn apply_row_col(b: &Bigram, row_col: &[String; 5], mode: &CipherMode) -> Option<Bigram> {
+    for rc in row_col.iter() {
+        if let Some(first) = rc.find(b.0) {
+            if let Some(second) = rc.find(b.1) {
+                let v: Vec<char> = rc.chars().collect();
+                match *mode {
+                    CipherMode::ENCRYPT => return Some((v[(first + 1) % 5], v[(second + 1) % 5])),
```

Comment:
Rather than using an enum to determine whether to encrypt/decrypt, it would be better if you passed in some sort of lambda - like I've done in the `affine` cipher when calling `substitute::shift_substitution`